tuf/api/metadata.py

```diff
@@ -1289,22 +1350,28 @@ class DelegatedRole(Role):
 
     def __init__(
         self,
-        name: str,
+        name: Optional[str],
         keyids: List[str],
         threshold: int,
         terminating: bool,
         paths: Optional[List[str]] = None,
         path_hash_prefixes: Optional[List[str]] = None,
+        succinct_hash_info: Optional[Dict[str, Any]] = None,
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
```

Comment:
strictly speaking this is still breaking API.  
`DelegatedRole("a", [], 1, False, [], None, {})`
used to be valid  code. Now it will fail I assume. 

New arguments should generally speaking be added in the end of the list... I'm not saying we can't break API here: it's not a major breakage by any means but maybe I'd still vote for adding the new arg in the end.

Reply:
I didn't add it to the end as in my mind `unrecognized_fields` are at the end of each class, but maybe that's not so important.